### PR TITLE
bug: Better error handling for Tokens page

### PR DESCRIPTION
**Pull request by mukulljangid**
## High Level Overview of Change
Add error messages for tokens page.

### Context of Change
Explorer wasn't getting any response from `rippled` because of an internal and response had `status: 200`, but data had the error.
Handling errors better.

### Type of Change

- [X] Bug fix (non-breaking change which fixes an issue)

## Before / After
Before:
image
After:
image

## Test Plan
Passes all CI tests

### CODE REVIEW

File: src/containers/Tokens/index.js [hunk 1]
```diff
@@ -25,6 +27,17 @@ const Tokens = props => {
       .get(TOP_TOKENS_URL)
       .then(res => {
         const { tokens } = res.data;
+
+        if (res.data.result === 'error') {
```
Comment (mukulljangid) [inline src/containers/Tokens/index.js:31]: I was thinking that we could keep this as !== success, for all random cases? @natenichols @mvadari

File: public/locales/ko-KP/translations.json [hunk 2]
```diff
@@ -196,5 +196,7 @@
   "invalid_ripple_address": "잘못된 Ripple 주소입니다.",
   "tokens": "토큰",
   "total_issuers": "총 발행자",
-  "total_tokens": "총 토큰"
+  "total_tokens": "총 토큰",
+  "error": "오류",
+  "internal error": "인터넷 서버 오류"
```
Comment (natenichols) [inline public/locales/ko-KP/translations.json:201]: Where did we get the translations?

File: src/containers/Tokens/index.js (hunk 1 above)
Comment (natenichols) [reply src/containers/Tokens/index.js:31]: `res?.data?.result !== 'success'` should handle cases where either res or data are `undefined`

Comment (mvadari) [reply src/containers/Tokens/index.js:31]: Doesn't really matter - the response is either `success` or `error` for this endpoint

File: src/containers/Tokens/index.js [hunk 3]
```diff
@@ -40,8 +53,13 @@ const Tokens = props => {
...
   }, []);
 
-  return error ? (
-    Tokens.renderError(error)
+  function renderError(errorInfo) {
+    const { result, message } = errorInfo;
+    return <NoMatch title={result} hints={[message]} />;
+  }
+
+  return error || isError ? (
```
Comment (natenichols) [inline src/containers/Tokens/index.js:61]: where is `error` defined here?

File: src/containers/Tokens/index.js (hunk 1 above)
Comment (mvadari) [reply src/containers/Tokens/index.js:31]: Up to you :)

File: src/containers/Tokens/index.js [hunk 4]
```diff
@@ -40,8 +53,13 @@ const Tokens = props => {
...
+    renderError(errorInfoRef.current)
```
Comment (mvadari) [inline src/containers/Tokens/index.js:62]: Why are we using a `ref` for this? We can just only render it if there's an error

File: src/containers/Tokens/index.js (hunk 1 above)
Comment (mukulljangid) [reply src/containers/Tokens/index.js:31]: > `res?.data?.result !== 'success'` should handle cases where either res or data are `undefined`

Yes, that's better

File: src/containers/Tokens/index.js (hunk 3 above)
Comment (mukulljangid) [reply src/containers/Tokens/index.js:61]: at the top, it is being deconstructed from props

**Review by natenichols — APPROVED** [review]

**Review by mvadari — APPROVED** [review]

**Pull request merged**